_posts/2018-10-25-Rust-1.30.md

```diff
@@ -0,0 +1,355 @@
+the crate root, but other paths would start at the local path, meaning:
+
+```rust
+mod foo {
+    pub fn bar() {
+        // ...
+    }
+}
+
+mod baz {
+    pub fn qux() {
+        // old
+        ::foo::bar();
+
+        // new
+        crate::foo::bar();
+    }
+}
+```
+
+This will hopefully make absolute paths a bit more clear, and remove some of
+the ugliness of leading `::`.
+
+All of these changes combined lead to a more straightforward understanding of
+how paths resolve. When you see a path like `a::b::c`, you can ask:
+
+* Is `a` the name of a crate? Then we're looking for `b::c` inside of it.
+* Is `a` the keyword `crate`? Then we're looking for `b::c` from the root of our crate.
+* Otherwise, we're looking for `a::b::c` from the current spot in the module hierarchy.
+
+Since these rules apply uniformly everywhere, you'll need to tweak your
+imports much less when moving code around.
+
+[usecrate]: https://github.com/rust-lang/rust/pull/54404/
+
+## Raw Identifiers
+
+[You can now use keywords as identifiers][rawidents] with some new syntax:
+
+```rust
+// define a local variable named `for`
+let r#for = true;
+
+// define a function named `for`
+fn r#for() {
+    // ...
+}
+
+// call that function
+r#for();
+```
+
+This doesn't have many use cases today, but will once you are trying to use a
+Rust 2015 crate with a Rust 2018 project, and vice-versa; we'll explain more
+in that upcoming blog post.
+
+[rawidents]: https://github.com/rust-lang/rust/pull/53236/
+
+## Other things
+
+Finally, you can [use the `#[panic_handler]`][panichandler] attribute to
+implement panics yourself, and you can now [match on visibility keywords,
+like `pub`, in macros][viskeyword] using the `vis` specifier.
+
+[panichandler]: https://github.com/rust-lang/rust/pull/51366/
+[viskeyword]: https://github.com/rust-lang/rust/pull/53370/
+
+See the [detailed release notes][notes] for more.
+
+### Library stabilizations
+
+A few new APIs were [stabilized for this
+release](https://github.com/rust-lang/rust/blob/master/RELEASES.md#stabilized-apis):
+
+* `Ipv4Addr::{BROADCAST, LOCALHOST, UNSPECIFIED}`
+* `Ipv6Addr::{BROADCAST, LOCALHOST, UNSPECIFIED}`
+* `Iterator::find_map`
+
+Additionally, the standard library has long had functions like `trim_left` to eliminate
+whitespace on one side of some text. However, when considering RTL languages, the meaning
+of "right" and "left" get confusing. As such, we're introducing some new names for these
+APIs:
+
+* `trim_left` -> `trim_start`
+* `trim_right` -> `trim_end`
+* `trim_left_matches` -> `trim_start_matches`
+* `trim_right_matches` -> `trim_end_matches`
+
+We plan to deprecate (but not removing, of course) the old names in Rust
+1.33.
+
+See the [detailed release notes][notes] for more.
+
+### Cargo features
+
+The largest feature of Cargo in this release is that we now [have a progress
+bar!](https://github.com/rust-lang/cargo/pull/5995/)
+
+> GIF goes here
```

Comment:
Just a reminder to put the gif here (or maybe asciinema?).

Reply:
I think a GIF is nicer as it starts automatically, so it is more eye-catching.